.dockerignore

```diff
@@ -1,4 +1,3 @@
-.git
```

Comment:
Is this only needed for the new `generated-files.Dockerfile` ?

Wondering if we need to add a `Dockerfile.dockerignore` so that the main Dockerfile doesn't get a cache-bust on any change, and so that we can specify different `.dockerignore` for each purpose.

Reply:
> Is this only needed for the new `generated-files.Dockerfile` ?

Yes only for `generated-files.Dockerfile`.

> Wondering if we need to add a `Dockerfile.dockerignore` so that the main Dockerfile doesn't get a cache-bust on any change, and so that we can specify different `.dockerignore` for each purpose.

Yes sounds good

Reply:
TBH; still eyeing the `tools.go` and if there's possible alternatives as well (separate module for the generator or something), but haven't looked closely yet.

Reply:
> Wondering if we need to add a `Dockerfile.dockerignore` so that the main Dockerfile.

Reminds me that we never implemented this for the classic builder, so the Windows dockerfile would still look at `.dockerignore`.

Perhaps we should look if we can easily implement `Dockerfile.dockerignore` on the CLI side for the classic builder 🤔

Reply:
Couldn't we create a temporary git repo inside the dockerfile?

Like:

```Dockerfile
COPY . .
RUN git init . && git add . && git commit -m"whatever"
RUN go generate && git ls-files ...
```